### PR TITLE
Upgrade the dependency version

### PR DESCRIPTION
**Pull request by Aisuko**
Signed-off-by: Aisuko <urakiny@gmail.com>

* This PR includes VSCODE config(.devcontainer) that can help the developer easier to develop the project. The image we use that includes all the environments that the project needs.

* As you guys know, I'd like to use more native way to import Linkerd2' functions and as I share the link I was discussed to  linekrd2 maintainer(https://github.com/linkerd/linkerd2/discussions/4467), I believe there may need more time to Linekrd2 more open. But, the dependency version we need to follow the Linkerd2 in order to keep compatible. If we won't do this, that could import the issue like `the package not include the xxxx packages`

### CODE REVIEW

**Comment by Aisuko** [comment]
@leecalcote  I believe that the PRs which includes the comment out protocol checks is not a good idea for our project

https://github.com/layer5io/meshery-linkerd/pull/73/checks?check_run_id=874192348

```
##[error]meshes/meshops.pb.go:495:9: xxx_messageInfo_EventsRequest.Unmarshal is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
82
##[error]meshes/meshops.pb.go:498:9: xxx_messageInfo_EventsRequest.Marshal is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
83
##[error]meshes/meshops.pb.go:501:2: xxx_messageInfo_EventsRequest.Merge is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
84
##[error]meshes/meshops.pb.go:504:9: xxx_messageInfo_EventsRequest.Size is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
85
##[error]meshes/meshops.pb.go:507:2: xxx_messageInfo_EventsRequest.DiscardUnknown is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
86
##[error]meshes/meshops.pb.go:510:35: proto.InternalMessageInfo is deprecated: Do not use; this type existed for intenal-use only.  (SA1019)
87
##[error]meshes/meshops.pb.go:529:9: xxx_messageInfo_EventsResponse.Unmarshal is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
88
##[error]meshes/meshops.pb.go:532:9: xxx_messageInfo_EventsResponse.Marshal is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
89
##[error]meshes/meshops.pb.go:535:2: xxx_messageInfo_EventsResponse.Merge is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
90
##[error]meshes/meshops.pb.go:538:9: xxx_messageInfo_EventsResponse.Size is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
91
##[error]meshes/meshops.pb.go:541:2: xxx_messageInfo_EventsResponse.DiscardUnknown is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
92
##[error]meshes/meshops.pb.go:544:36: proto.InternalMessageInfo is deprecated: Do not use; this type existed for intenal-use only.  (SA1019)
93
##[error]meshes/meshops.pb.go:575:2: proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
94
##[error]meshes/meshops.pb.go:576:2: proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
95
##[error]meshes/meshops.pb.go:577:2: proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
96
##[error]meshes/meshops.pb.go:578:2: proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
```

**Comment by leecalcote** [comment]
@Aisuko, clarifying... are you saying that you consider this [static check](https://github.com/layer5io/meshery-linkerd/blob/master/.github/workflows/ci.yml#L71) not ideal for the project and something we might want to disable?

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	
+	// Uncomment the next line to run commands after the container is created.
+	// "postCreateCommand": "go version",
+
+	// Comment out the next line to run as root
+	// "remoteUser": "vscode",
+	"remoteEnv": {
+		"GOPROXY":"https://goproxy.io,direct"
+	}
+}
```
Comment (leecalcote) [inline .devcontainer/devcontainer.json:32]: This devcontainer is pretty neat. I haven't seen this before.

**Comment by Aisuko** [comment]
> @Aisuko, clarifying... are you saying that you consider this [static check](https://github.com/layer5io/meshery-linkerd/blob/master/.github/workflows/ci.yml#L71) not ideal for the project and something we might want to disable?

I mean we should find a way to solve the issue to the `proto.xxxType` config deprecated
(Upgrading  and fix the issue ) rather than comment the static type check. If I remember right, we did this on meshery project

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (Aisuko) [reply .devcontainer/devcontainer.json:32]: Yep, https://github.com/Aisuko/dev-container  we can debug the project into the container(It also support the remote docker daemon)

**Pull request merged**